lib/posix/pthread.c

```diff
@@ -26,9 +37,29 @@ LOG_MODULE_REGISTER(pthread, CONFIG_PTHREAD_LOG_LEVEL);
 #define DYNAMIC_STACK_SIZE 0
 #endif
 
-#define _pthread_cancel_pos LOG2(PTHREAD_CANCEL_DISABLE)
+/* The maximum allowed stack size (for this implementation) */
+#define PTHREAD_STACK_MAX (UINT16_MAX + 1)
+
+static inline size_t __get_attr_stacksize(const struct pthread_attr *attr)
+{
+	return attr->stacksize + 1;
+}
+
+static inline void __set_attr_stacksize(struct pthread_attr *attr, size_t size)
```

Comment:
Should size be uint16?

Reply:
Maybe, but given that the argument is verified to be within range by either build time or runtime assertion before this is called, I'm not terribly worried about it.